staging/src/k8s.io/csi-translation-lib/plugins/gce_pd.go

```diff
@@ -60,6 +60,41 @@ func (g *gcePersistentDiskCSITranslator) TranslateInTreeStorageClassParametersTo
 	return scParameters, nil
 }
 
+// TranslateInTreeVolumeToCSI takes a Volume with GCEPersistentDisk set from in-tree
+// and converts the GCEPersistentDisk source to a CSIPersistentVolumeSource
+func (t *gcePersistentDiskCSITranslator) TranslateInTreeVolumeToCSI(volume *v1.Volume) (*v1.PersistentVolume, error) {
+	if volume == nil || volume.GCEPersistentDisk == nil {
+		return nil, fmt.Errorf("volume is nil or GCE PD not defined on volume")
+	}
+
+	pdSource := volume.GCEPersistentDisk
+
+	partition := ""
+	if pdSource.Partition != 0 {
+		partition = strconv.Itoa(int(pdSource.Partition))
+	}
+
+	pv := &v1.PersistentVolume{
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				CSI: &v1.CSIPersistentVolumeSource{
+					Driver: GCEPDDriverName,
+					// TODO: PDName is not going to work for GCE. We need to get this to a volIDZonalFmt
```

Comment:
Ah i understand the issue now, the format can be:
`volID = fmt.Sprintf(volIDZonalFmt, UnspecifiedValue, UnspecifiedValue, pv.Spec.GCEPersistentDisk.PDName)`

And what we can do is actually find the disk on the PD driver side when we encounter `UnspecifiedValue`.

we already do this here: https://github.com/davidz627/kubernetes/pull/5/files#diff-307e088a66bf6532855c7b623e41af8fR121

Reply:
Excellent!

Reply:
yep! And I just checked the PD CSI Driver code and looks like I predicted this and already implemented it already. https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/129

:D